### PR TITLE
move test num_images_per_prompt to pipeline mixin

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py [hunk 1]
```diff
@@ -671,7 +693,7 @@ def get_indices(self, prompt: str) -> Dict[str, int]:
     def __call__(
         self,
         prompt: Union[str, List[str]],
-        token_indices: List[int],
+        token_indices: Union[List[int], List[List[int]]],
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py:696]: One more list of list of ... and I am gone already hahaha.

**Review by sayakpaul — APPROVED** [review]
Clean 🔥

**Comment by williamberman** [comment]
If we merge https://github.com/huggingface/diffusers/pull/2502 , I can change these tests to use `batch_params`

**Comment by sayakpaul** [comment]
> If we merge #2502 , I can change these tests to use `batch_params`

Yeah worth waiting for that one to get merged meanwhile then.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py [hunk 2]
```diff
@@ -513,8 +513,30 @@ def check_inputs(
                     f" {negative_prompt_embeds.shape}."
                 )
 
-        if (indices is None) or (indices is not None and not isinstance(indices, List)):
-            raise ValueError(f"`indices` has to be a list but is {type(indices)}")
+        indices_is_list_ints = isinstance(indices, list) and isinstance(indices[0], int)
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py:516]: great!

**Review by yiyixuxu — APPROVED** [review]

**Review by patrickvonplaten — APPROVED** [review]
Super nice refactor! Thanks a lot!

File: tests/pipelines/unclip/test_unclip_image_variation.py [hunk 3]
```diff
@@ -362,58 +362,8 @@ def test_unclip_image_variation_input_list_images(self):
...
-                0.9997,
-                0.0023,
-                0.0029,
-                0.9997,
-                0.9985,
-                0.9997,
-                0.0010,
-                0.9995,
-            ]
-        )
```
Comment (pcuenca) [inline tests/pipelines/unclip/test_unclip_image_variation.py:413]: We lost this test, but I agree it's not a `num_images_per_prompt` test :)

**Review by pcuenca — APPROVED** [review]
Very cool!

**Comment by patrickvonplaten** [comment]
@williamberman happy to get this one in soon :-)

**Comment by williamberman** [comment]
> @williamberman happy to get this one in soon :-)

Yeah going to make some small changes because we merged the other PR with `batch_param` then merge today

**Pull request merged**